pkg/operator/defaultscccontroller/syncer.go

```diff
@@ -1,16 +1,71 @@
 package defaultscccontroller
 
 import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+
+	securityv1 "github.com/openshift/api/security/v1"
 	securityv1listers "github.com/openshift/client-go/security/listers/security/v1"
 	"github.com/openshift/library-go/pkg/operator/events"
-	"k8s.io/apimachinery/pkg/types"
 )
 
 type Syncer struct {
 	lister   securityv1listers.SecurityContextConstraintsLister
 	recorder events.Recorder
+	updater  OperatorStatusUpdater
+	cache    *DefaultSCCCache
 }
 
 func (s *Syncer) Sync(key types.NamespacedName) error {
-	return nil
+	// If it's not to do with the default SCC, we don't care.
+	if !s.cache.IsDefaultSCC(key.Name) {
+		return nil
+	}
+
+	mutated := make([]string, 0)
+	for _, name := range s.cache.DefaultSCCNames() {
+		original := s.cache.GetDefaultSCC(name)
+		if original == nil {
+			return fmt.Errorf("name=%s default scc not found in default scc cache", name)
+		}
+
+		current, err := s.lister.Get(name)
+		if err != nil {
+			if k8serrors.IsNotFound(err) {
+				klog.Infof("[%s] name=%s scc has been deleted - %s", ControllerName, name, err.Error())
+				continue
+			}
+
+			return err
+		}
+
+		// original can be mutated safely according to IsDefaultSCC, so no deep copy required.
+		copy1, copy2 := withNoMeta(current.DeepCopy(), original)
+		if !equality.Semantic.DeepEqual(copy1, copy2) {
+			mutated = append(mutated, name)
+		}
+	}
+
+	if len(mutated) > 0 {
+		klog.Infof("[%s] default scc has mutated %s", ControllerName, mutated)
+	}
+
+	condition := NewCondition(mutated)
+	return s.updater.UpdateStatus(condition)
+}
+
+func withNoMeta(obj1, obj2 *securityv1.SecurityContextConstraints) (copy1, copy2 *securityv1.SecurityContextConstraints) {
```

Comment:
having a func for two objects is strange. Call it twice with one.

Reply:
make the deepcopies internally here. `withNoMeta` with that signature does not suggest it has side-effects-